### ISSUE TITLE
Model.transform predict all result into 1.0

### ISSUE THREAD

**Issue by land1725** [intel-analytics/BigDL#2703]
@qiuxin2012 Hello,qiuxin:
I have found a new porblems.when I use Model.transform,it predict all result into 1.0
    val testRank = dlModel.transform(testInput.coalesce(4))
my model is two hidden layers ,first one is 256, second is 128,I use Adam ,learning rate is 1e-3, I use glorot_uniform method to init, the activation I use relu ,and betch size 1024,I trained 10 epoch.
my dataInput is about 185 size Vectors about 7000000 lines .
My code is like this :
val hidden_1 = Dense(outputDim = hidden1Number, activation = activationMethod, init = getInitMethod, inputShape = Shape(inputNumber))
    val hidden_2 = Dense(outputDim = hidden2Number, init = getInitMethod, activation = activationMethod)
    val hidden_3 = Dense(outputDim = 1, init = getInitMethod, activation = "sigmoid")

    val model = Sequential()
      .add(hidden_1)
      .add(hidden_2)
      .add(hidden_3)

    val criterion = BCECriterion()
    val estimator = new DLClassifier(model, criterion, Array(inputNumber))
      .setFeaturesCol("scaledFeatures")
      .setLabelCol("label")
      .setBatchSize(batchSize)
      .setMaxEpoch(maxEpoch)
      .setOptimMethod(optimMethod)
      .setValidation(Trigger.everyEpoch, testInput, Array(new AUC()), 1280)
    val dlModel = estimator.fit(trainInput.coalesce(4))
    val testRank = dlModel.transform(testInput.coalesce(4))
    testRank.show(1000, false)

**Comment by qiuxin2012** [comment]
It's a bug of DLclassifier, the DLClassifier doesn't support binary classification. I will fix it in our latest version.

You can use DLEstimator the work around this issue.
```
val estimator = new DLEstimator(model, criterion, Array(inputNumber), Array(1))
```
Then, the result colomn `prediction` will be a `WrappedArray[Double]`

**Comment by jason-dai** [comment]
I believe this is supported in NNClassifier (https://analytics-zoo.github.io/master/#APIGuide/PipelineAPI/nnframes/#nnclassifier) @hhbyyh ?

**Comment by hhbyyh** [comment]
Yes DLClassifier does not support binary classification with single model output. 

The function was added in NNClassifer in Analytics Zoo: https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/test/scala/com/intel/analytics/zoo/pipeline/nnframes/NNClassifierSpec.scala#L111

**Comment by qiuxin2012** [comment]
@jason-dai @hhbyyh 
NNClassifier throw exception as the feature's type is `org.apache.spark.ml.linalg.SparseVector`.
```
java.lang.IllegalArgumentException: SeqToTensor only supports Float, Double, Array[Float], Array[Double] or MLlib Vector but got (185,[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,26,29,32,34,37,40,42,43,77,78,79,80,81,82,83,84,85,86,87,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,154,161,168,171,173,175,176,177,178,179,180,181,182,183,184],[2.0020916998823446,0.5538819010600257,1.9620470855128804,0.6529908754355452,3.0224229134931972,1.1084939493655113,0.2222117208802306,1.15899232731025,0.547377716301749,1.529066978212077,2.2255396525617166,4.547091221734695,1.7635620819411306,11.588788143090822,2.1958626836750317,17.36319566682153,3.6223343829925687,2.8795067825067755,1.468694596258978,4.27376783158064,9.386158059183384,5.282768902209277,2.6980991822653713,0.2375538674160618,3.0699182002740115,2.1873633515504776,0.6055890571315301,0.7463637130870616,0.19232525503992082,0.6293778760426295,0.07253733889108296,0.30584513817151987,0.2065218529564285,3.6930035655938336,1.8820655802235475,3.1925839030373133,2.1751748230649213,4.312872004290533,2.64811440161769,3.88478655583379,2.861606723823091,2.2116581379633518,2.707714375258958,3.8056418924146764,1.8595032568267844,2.547305247917641,1.322905144262748,-0.6129328215479302,0.24225893287840206,6.246144306907719,-1.4801000384329965,2.4350563001023935,-2.7861033291319335,-2.7077147117694804,2.798277139062758,0.1827120877498848,0.24710310283142045,3.039058434224639,5.174203191908838,0.009423641836898627,0.011176444687454408,0.008493751072323178,0.02322086055108309,127.44195193253908,128.52185516237952,6.257172128560677,4.907474854847461,2.4417947974064353,2.5425496908054845,893.6503790633112,0.03681884602870073,0.12567276554590234,0.05722323223335646,0.20567492019245318,0.07326122190605218,0.07637574253430197,0.03937100622307364,0.1563531568736037,0.060714393709228265,-0.14615956901122468])
	at com.intel.analytics.zoo.feature.common.SeqToTensor$$anonfun$apply$1.apply(SeqToTensor.scala:39)
	at com.intel.analytics.zoo.feature.common.SeqToTensor$$anonfun$apply$1.apply(SeqToTensor.scala:31)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at com.intel.analytics.zoo.feature.common.FeatureLabelPreprocessing$$anonfun$apply$1.apply(FeatureLabelPreprocessing.scala:40)
	at com.intel.analytics.zoo.feature.common.FeatureLabelPreprocessing$$anonfun$apply$1.apply(FeatureLabelPreprocessing.scala:39)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:150)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:99)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:282)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2018-12-11 13:06:57 ERROR TaskSetManager:70 - Task 2 in stage 15.0 failed 1 times; aborting job
Exception in thread "main" org.apache.spark.SparkException: Job aborted due to stage failure: Task 2 in stage 15.0 failed 1 times, most recent failure: Lost task 2.0 in stage 15.0 (TID 300, localhost, executor driver): java.lang.IllegalArgumentException: SeqToTensor only supports Float, Double, Array[Float], Array[Double] or MLlib Vector but got (185,[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,26,29,32,34,37,40,42,43,77,78,79,80,81,82,83,84,85,86,87,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,154,161,168,171,173,175,176,177,178,179,180,181,182,183,184],[2.0020916998823446,0.5538819010600257,1.9620470855128804,0.6529908754355452,3.0224229134931972,1.1084939493655113,0.2222117208802306,1.15899232731025,0.547377716301749,1.529066978212077,2.2255396525617166,4.547091221734695,1.7635620819411306,11.588788143090822,2.1958626836750317,17.36319566682153,3.6223343829925687,2.8795067825067755,1.468694596258978,4.27376783158064,9.386158059183384,5.282768902209277,2.6980991822653713,0.2375538674160618,3.0699182002740115,2.1873633515504776,0.6055890571315301,0.7463637130870616,0.19232525503992082,0.6293778760426295,0.07253733889108296,0.30584513817151987,0.2065218529564285,3.6930035655938336,1.8820655802235475,3.1925839030373133,2.1751748230649213,4.312872004290533,2.64811440161769,3.88478655583379,2.861606723823091,2.2116581379633518,2.707714375258958,3.8056418924146764,1.8595032568267844,2.547305247917641,1.322905144262748,-0.6129328215479302,0.24225893287840206,6.246144306907719,-1.4801000384329965,2.4350563001023935,-2.7861033291319335,-2.7077147117694804,2.798277139062758,0.1827120877498848,0.24710310283142045,3.039058434224639,5.174203191908838,0.009423641836898627,0.011176444687454408,0.008493751072323178,0.02322086055108309,127.44195193253908,128.52185516237952,6.257172128560677,4.907474854847461,2.4417947974064353,2.5425496908054845,893.6503790633112,0.03681884602870073,0.12567276554590234,0.05722323223335646,0.20567492019245318,0.07326122190605218,0.07637574253430197,0.03937100622307364,0.1563531568736037,0.060714393709228265,-0.14615956901122468])
	at com.intel.analytics.zoo.feature.common.SeqToTensor$$anonfun$apply$1.apply(SeqToTensor.scala:39)
	at com.intel.analytics.zoo.feature.common.SeqToTensor$$anonfun$apply$1.apply(SeqToTensor.scala:31)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at com.intel.analytics.zoo.feature.common.FeatureLabelPreprocessing$$anonfun$apply$1.apply(FeatureLabelPreprocessing.scala:40)
	at com.intel.analytics.zoo.feature.common.FeatureLabelPreprocessing$$anonfun$apply$1.apply(FeatureLabelPreprocessing.scala:39)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:150)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:99)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:282)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1435)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1423)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1422)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1422)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:802)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:802)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:802)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1650)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1605)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1594)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:628)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1918)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1931)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1944)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1958)
	at org.apache.spark.rdd.RDD.count(RDD.scala:1157)
	at com.intel.analytics.bigdl.dataset.DistributedDataSet$$anon$5.cache(DataSet.scala:191)
	at com.intel.analytics.bigdl.optim.AbstractOptimizer.prepareInput(AbstractOptimizer.scala:253)
	at com.intel.analytics.bigdl.optim.DistriOptimizer.prepareInput(DistriOptimizer.scala:732)
	at com.intel.analytics.bigdl.optim.DistriOptimizer.optimize(DistriOptimizer.scala:784)
	at com.intel.analytics.zoo.pipeline.nnframes.NNEstimator.internalFit(NNEstimator.scala:399)
	at com.intel.analytics.zoo.pipeline.nnframes.NNEstimator.internalFit(NNEstimator.scala:163)
	at org.apache.spark.ml.DLEstimatorBase.fit(DLEstimatorBase.scala:58)
	at HuoyingBigDL$.main(HuoyingBigDL.scala:774)
	at HuoyingBigDL.main(HuoyingBigDL.scala)
Caused by: java.lang.IllegalArgumentException: SeqToTensor only supports Float, Double, Array[Float], Array[Double] or MLlib Vector but got (185,[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,26,29,32,34,37,40,42,43,77,78,79,80,81,82,83,84,85,86,87,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,154,161,168,171,173,175,176,177,178,179,180,181,182,183,184],[2.0020916998823446,0.5538819010600257,1.9620470855128804,0.6529908754355452,3.0224229134931972,1.1084939493655113,0.2222117208802306,1.15899232731025,0.547377716301749,1.529066978212077,2.2255396525617166,4.547091221734695,1.7635620819411306,11.588788143090822,2.1958626836750317,17.36319566682153,3.6223343829925687,2.8795067825067755,1.468694596258978,4.27376783158064,9.386158059183384,5.282768902209277,2.6980991822653713,0.2375538674160618,3.0699182002740115,2.1873633515504776,0.6055890571315301,0.7463637130870616,0.19232525503992082,0.6293778760426295,0.07253733889108296,0.30584513817151987,0.2065218529564285,3.6930035655938336,1.8820655802235475,3.1925839030373133,2.1751748230649213,4.312872004290533,2.64811440161769,3.88478655583379,2.861606723823091,2.2116581379633518,2.707714375258958,3.8056418924146764,1.8595032568267844,2.547305247917641,1.322905144262748,-0.6129328215479302,0.24225893287840206,6.246144306907719,-1.4801000384329965,2.4350563001023935,-2.7861033291319335,-2.7077147117694804,2.798277139062758,0.1827120877498848,0.24710310283142045,3.039058434224639,5.174203191908838,0.009423641836898627,0.011176444687454408,0.008493751072323178,0.02322086055108309,127.44195193253908,128.52185516237952,6.257172128560677,4.907474854847461,2.4417947974064353,2.5425496908054845,893.6503790633112,0.03681884602870073,0.12567276554590234,0.05722323223335646,0.20567492019245318,0.07326122190605218,0.07637574253430197,0.03937100622307364,0.1563531568736037,0.060714393709228265,-0.14615956901122468])
	at com.intel.analytics.zoo.feature.common.SeqToTensor$$anonfun$apply$1.apply(SeqToTensor.scala:39)
	at com.intel.analytics.zoo.feature.common.SeqToTensor$$anonfun$apply$1.apply(SeqToTensor.scala:31)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at com.intel.analytics.zoo.feature.common.FeatureLabelPreprocessing$$anonfun$apply$1.apply(FeatureLabelPreprocessing.scala:40)
	at com.intel.analytics.zoo.feature.common.FeatureLabelPreprocessing$$anonfun$apply$1.apply(FeatureLabelPreprocessing.scala:39)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:150)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:99)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:282)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
